### PR TITLE
Added streamline filtering by primary orientation; other bundle definition fixes

### PR DESCRIPTION
**Pull request by 36000**

### CODE REVIEW

**Comment by 36000** [comment]
@arokem  ready for review

File: AFQ/api/bundle_dict.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ def append_l_r(bundle_list, no_lr_list):
            "FA", "FP"]
 BUNDLES = append_l_r(BUNDLES, ["FA", "FP"])
 
+READ_BUNDLES = append_l_r(["pARC", "VOF"], [])
```
Comment (arokem) [inline AFQ/api/bundle_dict.py:34]: Shouldn't we make this part of our standard set?

File: AFQ/api/group.py [hunk 2]
```diff
@@ -511,6 +514,8 @@ def cmd_outputs(self, cmd="rm", dependent_on=None, exceptions=[]):
         for pAFQ in self.pAFQ_list:
             pAFQ.cmd_outputs(cmd, dependent_on, exceptions)
 
+    clobber = cmd_outputs
```
Comment (arokem) [inline AFQ/api/group.py:517]: This is just an alias for the default?

Comment (arokem) [reply AFQ/api/group.py:517]: If yes, a comment to that effect might be good to put here.

File: AFQ/api/participant.py [hunk 3]
```diff
@@ -235,12 +237,22 @@ def cmd_outputs(self, cmd="rm", dependent_on=None, exceptions=[]):
...
             elif os.path.isdir(full_path):
                 # other than ROIs, folders are dependent on everything
                 if dependent_on is None or filename != "ROIs":
                     os.system(f"{cmd} -r {full_path}")
+
+        # do not assume previous calculations are still valid
+        # after file operations
+        self.make_workflow()
+
+    clobber = cmd_outputs
```
Comment (arokem) [inline AFQ/api/participant.py:258]: And same here.

File: AFQ/definitions/image.py [hunk 4]
```diff
@@ -229,7 +229,12 @@ class RoiImage(ImageDefinition):
     api.GroupAFQ(tracking_params={"seed_image": seed_image})
     """
 
-    def __init__(self, use_presegment=False, use_endpoints=False):
+    def __init__(self,
+                 use_waypoints=True,
+                 use_presegment=False,
+                 use_endpoints=False):
+        # TODO: assert at least one is true
```
Comment (arokem) [inline AFQ/definitions/image.py]: Still a TODO? To add in this PR? (I vote yes)

File: AFQ/tests/test_bundle_dict.py [hunk 5]
```diff
@@ -52,13 +52,12 @@ def test_BundleDict():
...
-    # not included and does not exist in afq templates
+    # mispelled bundle that does not exist in afq templates
     with pytest.raises(
             ValueError,
-            match="VOF_L is not in AFQ templates"):
-        afq_bundles = abd.BundleDict(["VOF_L", "VOF_R"])
-        afq_bundles["VOF_R"]
+            match=" is not in AFQ templates"):
+        afq_bundles = abd.BundleDict(["VOQ_L", "VOQ_R"])
+        afq_bundles["VOQ_R"]
```
Comment (arokem) [inline AFQ/tests/test_bundle_dict.py:60]: Nice

File: AFQ/tractography.py [hunk 6]
```diff
@@ -21,7 +21,7 @@
 def track(params_file, directions="prob", max_angle=30., sphere=None,
           seed_mask=None, seed_threshold=0, n_seeds=1, random_seeds=False,
           rng_seed=None, stop_mask=None, stop_threshold=0, step_size=0.5,
-          min_length=10, max_length=1000, odf_model="CSD",
+          min_length=40, max_length=200, odf_model="CSD",
```
Comment (arokem) [inline AFQ/tractography.py:24]: These are pretty dramatic changes! Can you say a bit more about this?

File: AFQ/segmentation.py [hunk 7]
```diff
@@ -1184,6 +1243,43 @@ def _is_streamline_in_ROIs_parallel(indiv_args, tol, include_roi,
             streamlines_in_bundles)
 
 
+def clean_by_orientation(streamlines, primary_axis, tol=None):
```
Comment (arokem) [inline AFQ/segmentation.py:1246]: I think that it would be great to have a unit test for this function

File: AFQ/segmentation.py [hunk 8]
```diff
@@ -1184,6 +1243,43 @@ def _is_streamline_in_ROIs_parallel(indiv_args, tol, include_roi,
...
+    """
+    Compute the cardinal orientation of each streamline
+    Parmaeters
+    __________
+    streamlines : sequence of 3XN_i arrays
```
Comment (arokem) [inline AFQ/segmentation.py:1251]: Need to finish this docstring

Comment (arokem) [reply AFQ/segmentation.py:1251]: What is `N_i`?

**Review by arokem — COMMENTED** [review]
Main comment is whether we should make these part of the default. Why not?

File: AFQ/tractography.py (hunk 6 above)
Comment (36000) [reply AFQ/tractography.py:24]: This should bring us back towards mAFQ defaults. I don't remember why we chose 10 and 1000 in the first place, but Jason Maya and I had a conversation about these and in most cases these lengths will serve better. One can adjust min length if they need to. The smaller streamlines were also making cleaning harder by changing the distribution.

Comment (arokem) [reply AFQ/tractography.py:24]: That makes sense! Thanks for clarifying.

**Pull request merged**